test.sh

```diff
@@ -1,51 +1,92 @@
 #!/bin/bash -e
-# test suite...
-echo running test.sh
-echo "ENV:"
-env
+# runs all (or selected) test suite(s) in test/ and aggregates results
+# Usage:
+#	./test.sh
+#	./test.sh gofmt
 
-failures=''
-function run-test()
+# library of utility functions
+# shellcheck disable=SC1091
+. test/util.sh
+
+# allow specifying a single testsuite to run
+testsuite="$1"
+
+# print environment when running all testsuites
+test -z "$testsuite" && (echo "ENV:"; env; echo; )
+
+# run a test and record failures
+function run-testsuite()
 {
-	$@ || failures=$( [ -n "$failures" ] && echo "$failures\\n$@" || echo "$@" )
+	testname="$(basename "$1" .sh)"
+	# if not running all test or this test is not explicitly selected, skip it
+	if test -z "$testsuite" || test "test-$testsuite" = "$testname";then
+		$@ || failures=$( [ -n "$failures" ] && echo "$failures\\n$@" || echo "$@" )
+	fi
 }
 
-# ensure there is no trailing whitespace or other whitespace errors
-run-test git diff-tree --check $(git hash-object -t tree /dev/null) HEAD
+# only run test if it is explicitly selected, otherwise report it is skipped
+function skip-testsuite()
+{
+	testname=$(basename "$1" .sh)
+	# show skip message only when running full suite
+	if test -z "$testsuite";then
+		echo skipping "$@" "($REASON)"
+		blue SKIP
+	else
+		# if a skipped suite is explicity called, run it anyway
+		if test "test-$testsuite" == "$testname";then
+			run-testsuite "$@"
+		fi
+	fi
+}
 
-# ensure entries to authors file are sorted
-start=$(($(grep -n '^[[:space:]]*$' AUTHORS | awk -F ':' '{print $1}' | head -1) + 1))
-run-test diff <(tail -n +$start AUTHORS | sort) <(tail -n +$start AUTHORS)
+# used at the end to tell if everything went fine
+failures=''
 
-run-test ./test/test-gofmt.sh
-run-test ./test/test-yamlfmt.sh
-run-test ./test/test-bashfmt.sh
-run-test ./test/test-headerfmt.sh
-run-test ./test/test-commit-message.sh
-run-test ./test/test-govet.sh
-run-test ./test/test-examples.sh
-run-test ./test/test-gotest.sh
+run-testsuite ./test/test-misc.sh
+run-testsuite ./test/test-gofmt.sh
+run-testsuite ./test/test-yamlfmt.sh
+run-testsuite ./test/test-bashfmt.sh
+run-testsuite ./test/test-headerfmt.sh
+run-testsuite ./test/test-commit-message.sh
+run-testsuite ./test/test-govet.sh
+run-testsuite ./test/test-gotest.sh
```

Comment:
I'm okay that you changed the order on these last two, it probably makes more sense, but was it a conscious decision, and if so, why.

Reply:
Good you question this! gofmt was intentional because it is quick and might give you something to work on already while waiting for the rest of the suite to complete, gotest/examples was not intentional, i'll reverse them back